include/Common.h

```diff
@@ -16,6 +19,25 @@ namespace Common {
 
     int RandomInt(int min, int max);
 
+    template <typename T>
+    T RandomEvent(std::vector<int> weights, std::vector<T> outcomes) {
+        std::vector <int> events;
```

Comment:
* rename to something like `indices`, since this vector will contain a bunch of indices